osxcollector/osxcollector.py

```diff
@@ -1512,12 +1520,14 @@ def ignore_db_path(sqlite_db_path):
             return sqlite_db_path.endswith('-journal') or os.path.isdir(
                 sqlite_db_path)
 
-        self._log_directories_of_dbs(
-            directories_of_dbs, chrome_path, chrome_ignored_sqlite_keys,
-            ignore_db_path)
+        for profile_path in profile_paths:
+            self._log_directories_of_dbs(
+                directories_of_dbs, profile_path, chrome_ignored_sqlite_keys,
+                ignore_db_path)
 
-        with Logger.Extra('osxcollector_subsection', 'preferences'):
-            self._log_json_file(chrome_path, 'preferences')
+        for profile_path in profile_paths:
```

Comment:
It looks like you have introduced 3 new for loops, each looping over the same resource. Any way you can refactor this code to merge it into 1 loop?